app/jinja_filters.py

```diff
@@ -453,9 +453,11 @@ def __init__(  # noqa: C901, R0912 pylint: disable=too-complex, too-many-branche
             and answer["label"]
         ):
             self.rowTitle = answer["label"]
+            self.id = answer["id"] + "-label"
             self.rowTitleAttributes = {"data-qa": answer["id"] + "-label"}
         else:
             self.rowTitle = strip_tags(question["title"])
+            self.id = question["id"] + "-row-item"
```

Comment:
The IDs shouldn't be suffixed. Also in both cases, we want to use the answer id but we only really need to set the answer id, if it is a row that can be edited, so:
```py
        if edit_link_text:
            self.id = answer["id"]
```